micromamba/src/run.cpp

```diff
@@ -161,6 +161,8 @@ namespace mamba
                        std::unique_ptr<LockFile> proc_dir_lock = lock_proc_dir())
             : location{ proc_dir() / fmt::format("{}.json", getpid()) }
         {
+            if (Context::instance().disable_lockfile)
```

Comment:
I think the code modified here does not do what you think. IIUC this creates a "pid file" that is used to give a list of running processes (see `~/.mamba/proc/` and `micromamba ps`)

Reply:
I agree and I originally did not have this line, however on my older hpc I was still getting the error:
```
mamba run' failed to open/create file...
```

I cannot explain this, but when I disable the json pid fifles from being created - everything works great.

Reply:
@danpf is the problem here with the lockfile or generally with writing to the home directory?

We do in fact lock the PID directory while the process is writing out the json file, so it's possible that it's also a lockfile issue.

Would it make more sense to put these checks in the `lock_proc_dir()` function?